.github/workflows/lint.yml

```
@@ -0,0 +1,30 @@
+name: Lint Check
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Source checkout
+        uses: actions/checkout@v4
+      - name: Set up Node 20
+        uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - name: Dependency restore
+        id: cache-restore
+        uses: actions/cache@v4
+        with:
+          path: node_modules
+          key: js-depend-${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+      - name: Dependency installing
+        if: steps.cache-restore.outputs.cache-hit != 'true'
+        run: npm i
```

dependencies treeはlockfileからそのまま復元させると思われるので、`npm install` ではなく `npm ci`

```suggestion
        run: npm ci
```